pkgs/shells/zsh/default.nix

```diff
@@ -64,34 +64,34 @@ stdenv.mkDerivation {
   postInstall = ''
     make install.info install.html
     mkdir -p $out/etc/
-    cat > $out/etc/zprofile <<EOF
+    cat > $out/etc/zshenv <<EOF
 if test -e /etc/NIXOS; then
-  if test -r /etc/zprofile; then
-    . /etc/zprofile
+  if test -r /etc/zshenv; then
+    . /etc/zshenv
   else
     emulate bash
     alias shopt=false
     . /etc/profile
```

Comment:
My motivation of this pr is to solve the issue that env vars set in `.zshenv` are overridden by those in `/etc/set-environment` (`$out/etc/zprofile` sources `/etc/profile` which sources `/etc/set-environment`) on NixOS system with `programs.zsh.enable = false`. This issue comes from users of home-manager.

I can think of three proposals to deal with this issue:

Proposal 0: souce `/etc/profile` in `$out/etc/zprofile`, i.e., the old behaviour
  - Pros
    - `config.environment.loginShellInit` is sourced in login zsh
  - Cons
    - env vars set in `.zshenv` are overridden by those in `/etc/set-environment`
	- `config.environment.shellInit` is not sourced in non-login zsh
    - `config.programs.bash.shellInit` is **also** sourced in login zsh
    - `config.programs.bash.loginShellInit` is **also** sourced in login zsh

Proposal 1: souce `/etc/profile` in `$out/etc/zshenv`, i.e., what this patch currently dose
  - Pros
    - env vars set in `.zshenv` are not overridden by those in `/etc/set-environment`
    - `config.environment.shellInit` is sourced in **all** zsh
  - Cons
    - `config.environment.loginShellInit` is **also** sourced in non-login zsh
    - `config.programs.bash.shellInit` is **also** sourced in **all** zsh
    - `config.programs.bash.loginShellInit` is **also** sourced in **all** zsh

Proposal 2: souce `/etc/set-environment` in `$out/etc/zshenv`
  - Pros
    - env vars set in `.zshenv` are not overridden by those in `/etc/set-environment`
    - `config.programs.bash.shellInit` is **not** sourced in **any** zsh
    - `config.programs.bash.loginShellInit` is **not** sourced in **any** zsh
  - Cons
    - `config.environment.shellInit` is **not** sourced in **any** zsh
    - `config.environment.loginShellInit` is **not** sourced in **any** zsh

I prefer proposal 2 because it fixes the issue without any wrong behaviour, i.e., sourcing config of bash. If you want those shell configs in `config.environment`, just set `programs.zsh.enable`.

WDYT?

Reply:
I also think proposal 2 sounds more reasonable.

Reply:
I have changed to proposal 2.